### ISSUE TITLE
RCS1197 does not know about `AppendInterpolatedStringHandler`

### ISSUE THREAD

**Issue by jnyrup** [dotnet/roslynator#899]
**Product and Version Used**: 
v4.1.0

**Steps to Reproduce**:
C# 10 introduced [Improved Interpolated Strings](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/improved-interpolated-strings) and .NET 6 added an overload `StringBuilder.Append(ref AppendInterpolatedStringHandler)` which lets you write
```cs
new StringBuilder().Append($"13{37}");
```

and at compile time it is translated into 

```cs
StringBuilder stringBuilder = new StringBuilder();
StringBuilder.AppendInterpolatedStringHandler handler = new StringBuilder.AppendInterpolatedStringHandler(2, 1, stringBuilder);
handler.AppendLiteral("13");
handler.AppendFormatted(37);
stringBuilder.Append(ref handler);
```

[SharpLab](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQAcoKgW1QOzgBsiBjACwlIGsBLfAcwB8ABABgAIWBGAOgBUIADxgBuALAAoFgGYuAJg4BhKQG8pHTVzksALBwDKAexwQAshBjkjAEwAUASg1b1kre474IAd0MwE9AwAQnC0RDYQCI68AILo6BD49gAkAETcMqoyAOwAvqkOEm5auVK5QA===)

![image](https://user-images.githubusercontent.com/919634/165911078-5bde3c71-02a0-438e-8e22-affdf0b321fd.png)
![image](https://user-images.githubusercontent.com/919634/165911101-ae9dc217-93aa-4c62-a5fc-c3381a9ea955.png)

**Actual Behavior**:
RCS1197 is reported

**Expected Behavior**:
RCS1197 should not be reported

**Comment by fgimian** [comment]
Confirmed here also, and I think that:

```c#
arguments.Append($" CONTENTDIR=\"{destinationPath}\"");
```

Really does look much clearer than:

```c#
arguments.Append(" CONTENTDIR=\"").Append(destinationPath).Append('\"');
```

😸

**Comment by udlose** [comment]
any update on this bug?

**Comment by josefpihrt** [comment]
> Confirmed here also, and I think that:
> 
> ```cs
> arguments.Append($" CONTENTDIR=\"{destinationPath}\"");
> ```
> 
> Really does look much clearer than:
> 
> ```cs
> arguments.Append(" CONTENTDIR=\"").Append(destinationPath).Append('\"');
> ```
> 

I agree.

Let's remove cases with interpolated string and string concatenation from RCS1197.
